client/test_commands/test_altbn128_mixer_base.py

```diff
@@ -0,0 +1,148 @@
+    2)
+
+HTAGS = [
+    int(
+        "1100000000000000000000000000000000000000000000000000000000000000"
+        "0000000000000000000000000000000000000000000000000000000000000000"
+        "0000000000000000000000000000000000000000000000000000000000000000"
+        "0000000000000000000000000000000000000000000000000000000000110010",
+        2),
+    int(
+        "1110000000000000000000000000000000000000000000000000000000000000"
+        "0000000000000000000000000000000000000000000000000000000000000000"
+        "0000000000000000000000000000000000000000000000000000000000000000"
+        "0000000000000000000000000000000000000000000000000000000000111011",
+        2),
+]
+
+VPUB = (0x5555555555555500, 0x00eeeeeeeeeeeeee)
+
+# 255                                         128         64           0
+# |<empty>|<h_sig>|<nullifiers>|<msg_auth_tags>|<v_pub_in>)|<v_pub_out>|
+RESIDUAL_BITS = int(
+    "101"  # h_sig
+    "010"  # nf_1
+    "001"  # nf_0
+    "111"  # htag_1
+    "110"  # htag_0
+    "0101010101010101010101010101010101010101010101010101010100000000"  # vin
+    "0000000011101110111011101110111011101110111011101110111011101110",  # vout
+    2)
+
+PACKED_PRIMARY_INPUTS = \
+    [ROOT] + COMMITMENTS + NULLIFIERS + [HSIG] + HTAGS + [RESIDUAL_BITS]
+
+
+def test_assemble_nullifiers(mixer_instance: Any) -> None:
+    # Test retrieving nullifiers
+    print("--- testing ", "test_assemble_nullifiers")
+    for i in range(JS_INPUTS):
+        res = mixer_instance.functions.\
+            assemble_nullifier_test(i, PACKED_PRIMARY_INPUTS).call()
+        val = int.from_bytes(res, byteorder="big")
+        assert val == NULLIFIERS[i], f"expected: {NULLIFIERS[i]}, got: {val}"
+
+
+def test_assemble_hsig(mixer_instance: Any) -> None:
+    # Test retrieving commitments
+    print("--- testing ", "test_assemble_hsig")
+    res = mixer_instance.functions.\
+        assemble_hsig_test(PACKED_PRIMARY_INPUTS).call()
+    hsig = int.from_bytes(res, byteorder="big")
+    assert hsig == HSIG, f"expected: {HSIG}, got {hsig}"
+
+
+def test_assemble_vpub(mixer_instance: Any) -> None:
+    # Test retrieving commitments
+    print("--- testing ", "test_assemble_vpub")
+    v_in, v_out = mixer_instance.functions.assemble_public_values_test(
+        PACKED_PRIMARY_INPUTS[-1]).call()
+    v_in_expect = VPUB[0] * ZETH_PUBLIC_UNIT_VALUE
+    v_out_expect = VPUB[1] * ZETH_PUBLIC_UNIT_VALUE
+    assert v_in == v_in_expect, f"expected: {v_in_expect}, got: {v_in}"
+    assert v_out == v_out_expect, f"expected: {v_out_expect}, got: {v_out}"
+
+
+def main() -> None:
+    print("Deploying AltBN128MixerBase_test.sol")
+    _web3, eth = mock.open_test_web3()
+    deployer_eth_address = eth.accounts[0]
+    _mixer_interface, mixer_instance = mock.deploy_contract(
+        eth,
+        deployer_eth_address,
+        "AltBN128MixerBase_test",
+        {
+            'mk_depth': ZETH_MERKLE_TREE_DEPTH,
+        })
```

Comment:
What about having a linebreak before closing the function call here?

Reply:
We could switch to that style, but I think most existing code keeps the closign parenthesis on the same line.

Reply:
Ah right, let's stay consistent with existing code then.